Amazon.Ion.ObjectMapper/IonDictionarySerializer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Amazon.IonDotnet;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Amazon.Ion.ObjectMapper
+{
+    public class IonDictionarySerializer : IonSerializer<IDictionary>
+    {
+        private IonSerializer serializer;
+        private Type valueType;
+
+        public IonDictionarySerializer(IonSerializer ionSerializer, Type valueType)
+        {
+            this.serializer = ionSerializer;
+            this.valueType = valueType;
+        }
+
+        public override IDictionary Deserialize(IIonReader reader)
+        {
+            reader.StepIn();
+
+            Type typedDictionaryType = typeof(Dictionary<,>).MakeGenericType(typeof(string), this.valueType);
+            MethodInfo methodInfo = typedDictionaryType.GetMethod("Add");
+            var dictionary = Activator.CreateInstance(typedDictionaryType);
```

Comment:
`dictionary` should be an actual IDictionary at this point, right? So we shouldn't need to use reflection to invoke the `Add` method?

Reply:
This Add method is for the generic typed `Dictionary<K, V>` which `IDictionary` does not have.

Reply:
I don't understand this comment, see https://docs.microsoft.com/en-us/dotnet/api/system.collections.idictionary.add?view=net-5.0 which shows `IDictionary` having an `Add` method. Regardless, given that `dictionary` is of a type if `IDictionary` (generic or not), both will have an `Add` that should be directly invokable?

Reply:
AH true I see what you meant now. Optimized and updated.